packages/bitcore-client/src/encryption.ts

```diff
@@ -59,6 +62,80 @@ function decryptPrivateKey(encPrivateKey, pubKey, encryptionKey) {
   return decrypted;
 }
 
+function sha512KDF(passphrase, salt, derivationOptions) {
+  let rounds =  derivationOptions.rounds || 1;
+  // if salt was sent in as a string, we will have to assume the default encoding type
+  if (!Buffer.isBuffer(salt)) {
+    salt = new Buffer(salt, 'hex');
+  }
+  let derivation = Buffer.concat([new Buffer(''), new Buffer(passphrase), salt]);
+  for (let i = 0; i < rounds; i++) {
+    derivation = crypto.createHash('sha512').update(derivation).digest();
+  }
+  return derivation;
+};
+
+function scryptKDF(passphrase, salt, derivationOptions) {
+  let opts = _.assign({ N: Math.pow(2, 14), r: 8, p: 8 }, derivationOptions);
+  return scrypt.hash(passphrase, opts, 48, salt);
+};
+
+export function bitcoinCoreDecrypt(jsonl, passphrase) {
+  const derivationMethods = { SHA512: 0 };
+  let master = null;
+  let jsonlDecrypted = [];
+  for (let line of jsonl) {
+    let cipherText = line.cipherText;
+    if (line.derivationMethod) {
+      let salt = line.salt;
+      let derivationOptions = {
+        method: derivationMethods[line.derivationMethod],
+        rounds: line.rounds
+      }
+      let hashFunc = hashPassphrase(derivationOptions);
+      let key = hashFunc(passphrase, salt, derivationOptions);
+      master = decrypt({cipherText, salt, key });
+    } else {
+      let privKey = decrypt({
+        key: master,
+        iv: bitcore.crypto.Hash.sha256sha256(Buffer.from(line.pubKey, 'hex')),
+        cipherText
+      });
+      let keyObj = {
+        privKey: privKey,
+        pubKey: line.pubKey,
+        address: line.address
+      };
+      jsonlDecrypted.push(keyObj);
+    }
+  }
+  return { jsonlDecrypted, master };
```

Comment:
Having a return statement on the same indention level as these function definitions weird

Reply:
this has been updated, moved the functions outside